ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -286,6 +286,40 @@ private boolean rekeyRequired(List<Expression> groupByExpressions) {
         && fieldNameFromExpression(groupByExpressions.get(0)).equals(keyFieldName));
   }
 
+  protected static Pair<String, List<Integer>> keyIndexesForGroupBy(
+      final Schema schema, final List<Expression> groupByExpressions) {
+    final StringBuilder aggregateKeyName = new StringBuilder();
+    final List<Integer> newKeyIndexes = new ArrayList<>();
+
+    // Collect the column indexes, and build the new key as <column1>+<column2>+...
+    boolean addSeparator = false;
+    for (Expression groupByExpr : groupByExpressions) {
+      if (addSeparator) {
+        aggregateKeyName.append("|+|");
```

Comment:
Now that we are using `|+|` in more than one place maybe yo can make it a final static variable.